### PR TITLE
Only lint files in workspace

### PR DESCRIPTION
**Pull request by ZacLN**
Adds rootPath to LanguageServer type

### CODE REVIEW

**Comment by ZacLN** [comment]
@davidanthoff, I'm unsure of the rules we actually want for this. For example I'd want  to allow this for all files

File: scripts/languageserver/provider_misc.jl [hunk 1]
```diff
@@ -10,6 +10,7 @@ const serverCapabilities = ServerCapabilities(
                         SignatureHelpOptions(["("])) 
 
 function process(r::JSONRPC.Request{Val{Symbol("initialize")},Dict{String,Any}}, server)
+    server.rootPath=haskey(r.params,"rootPath") ? "file://"*r.params["rootPath"] : ""
```
Comment (davidanthoff) [inline scripts/languageserver/provider_misc.jl]: Maybe not use the ``file://`` prefix here, after all this is not a file URI? Maybe just store the path as we get it?

File: scripts/languageserver/provider_misc.jl [hunk 2]
```diff
@@ -21,7 +22,9 @@ end
 function process(r::Request{Val{Symbol("textDocument/didOpen")},DidOpenTextDocumentParams}, server)
     server.documents[r.params.textDocument.uri] = split(r.params.textDocument.text,r"\r\n?|\n")
     
-    process_diagnostics(r.params.textDocument.uri, server)
+    if r.params.textDocument.uri[1:min(length(r.params.textDocument.uri),length(server.rootPath))] == server.rootPath
```
Comment (davidanthoff) [inline scripts/languageserver/provider_misc.jl]: I'm wondering what the right behavior is when ``rootPath`` is ``""``... We probably still want to lint in that case? Maybe we actually need to rules: 1) if a folder is open, only lint files from that folder, 2) if no folder is open, lint any file, except files from the base library? I'm not sure, though...

**Comment by ZacLN** [comment]
couldn't fix conflicts so I've opened a new one.

**Pull request closed without merging**